packages/react-paypal-js/src/v6/hooks/usePayLaterOneTimePaymentSession.ts

```diff
@@ -30,17 +31,26 @@ export function usePayLaterOneTimePaymentSession({
     ...callbacks
 }: PayLaterOneTimePaymentSessionProps): BasePaymentSessionReturn {
     const { sdkInstance } = usePayPal();
+    const isMountedRef = useIsMountedRef();
     const sessionRef = useRef<OneTimePaymentSession | null>(null); // handle cleanup
     const proxyCallbacks = useProxyProps(callbacks);
+    const [error, setError] = useState<Error | null>(null);
 
     const handleDestroy = useCallback(() => {
         sessionRef.current?.destroy();
         sessionRef.current = null;
     }, []);
 
+    // Separate error reporting effect to avoid infinite loops with proxyCallbacks
```

Comment:
IMO a bit of a bummer that we need this separate effect, but I'm not sure another way we can prevent an infinite re-render loop with `proxyProps` and `setError` if a merchant passes in a new function for every render of one of these hooks.